pypeit/core/coadd.py

```diff
@@ -1294,67 +1294,234 @@ def order_phot_scale(spectra, phot_scale_dicts, nsig=3.0, niter=5, debug=False):
+                plt.xlim([np.min(wave[allok_iord_ref]), np.max(wave[allok_iord_scl])])
                 plt.legend()
                 plt.xlabel('wavelength')
                 plt.ylabel('Flux')
                 plt.show()
         else:
-            msgs.warn('Not enough overlap region for sticking different orders.')
+            msgs.warn('Not enough spectral overlap to rescale spectra in order {:d}.'.format(iord) + ' Not recaling for this order'
+                      'Consider decreasing min_overlap_frac = {:5.3f}'.format(min_overlap_frac))
+            fluxes_out = fluxes_in[iord-1,:]
+            sigs_out = sigs_in[iord-1,:]
+
+
+
+def merge_order(spectra, wave_grid, extract='OPT', orderscale='median', niter=5, sigrej_final=3., SN_MIN_MEDSCALE = 5.0,
+                overlapfrac = 0.01, num_min_pixels=10,phot_scale_dicts=None, qafile=None, outfile=None, debug=False):
+    """
+        routines for merging orders of echelle spectra.
+    parameters:
+        spectra: spectra in the format of Xspectrum1D
+        wave_grid (numpy array): The final wavelength grid for the output. Note that the wave_grid should be
+          consistent with your spectra, if not please rebin it first.
+        extract (str): 'OPT' or 'BOX'
+        orderscale (str): which method you want use to scale different orders: median, photometry or None.
+        niter (int): number of iteration for rejections in the overlapped part
+        sigrej_final (float): sigma for rejections in the overlapped part
+        SN_MIN_MEDSCALE (float): minimum SNR for scaling different orders
+        overlapfrac (float): minimum overlap fraction for scaling different orders.
+        phot_scale_dicts (dict): A dict contains photometric information of each order. An example is given below.
+          phot_scale_dicts = {'0': {'filter': None, 'mag': None, 'mag_type': None, 'masks': None},
+                              '1': {'filter': 'UKIRT-Y', 'mag': 20.33, 'mag_type': 'AB', 'masks': None},
+                              '2': {'filter': 'UKIRT-J', 'mag': 20.19, 'mag_type': 'AB', 'masks': None},
+                              '3': {'filter': 'UKIRT-H', 'mag': 20.02, 'mag_type': 'AB', 'masks': None},
+                              '4': {'filter': 'UKIRT-K', 'mag': 19.92, 'mag_type': 'AB', 'masks': None}}
+        qafile (str): name of qafile
+        outfile (str): name of file to be saved for your final spectrum
+        debug (bool): show debug plots?
+    returns:
+        spec1d: XSpectrum1D after order merging.
+    """
+
+    ## Scaling different orders
+    if orderscale == 'photometry':
+        # Only tested on NIRES.
+        if phot_scale_dicts is not None:
+            spectra = order_phot_scale(spectra, phot_scale_dicts, debug=debug)
+        else:
+            msgs.warn('No photometric information is provided. Will use median scale.')
+            orderscale = 'median'
+    elif orderscale == 'median':
```

Comment:
If `orderscale` gets changed from photometry to median, this `elif` won't catch it.  You can change it to `if`, and then replace the `else` with `if orderscale not in ['photometry', 'median']: `

Reply:
I was hoping we can address this before we merge.

Reply:
@feigewang can you comment. 

Anyway this stuff is under construction. Nobody is using it except developers.

Reply:
Changed according to Kyle's comments. Thanks!